src/modules/job-manager/plugins/limit-duration.c

```diff
@@ -0,0 +1,323 @@
+                            "duration", &o) < 0) {
+        errprintf (error, "%s: %s", name, jerror.text);
+        return -1;
+    }
+    if (o) {
+        if (json_is_string (o)) {
+            if (fsd_parse_duration (json_string_value (o), &d) < 0) {
+                errprintf (error, "%s: FSD value is malformed", name);
+                return -1;
+            }
+        }
+        else if (json_is_number (o)) {
+            if ((d = json_real_value (o)) < 0) {
+                errprintf (error, "%s: must be >= 0", name);
+                goto inval;
+            }
+        }
+        else {
+            errprintf (error, "%s: must be FSD (string) or seconds", name);
+            goto inval;
+        }
+    }
+    *duration = d;
+    return 0;
+inval:
+    errno = EINVAL;
+    return -1;
+}
+
+static int queues_parse (zhashx_t **zhp,
+                         json_t *conf,
+                         flux_error_t *error)
+{
+    json_t *queues;
+    const char *name;
+    json_t *entry;
+    double duration;
+    flux_error_t e;
+    zhashx_t *zh;
+
+    if (!(zh = queues_create ())) {
+        errprintf (error, "out of memory parsing [queues]");
+        goto error;
+    }
+    if ((queues = json_object_get (conf, "queues"))) {
+        json_object_foreach (queues, name, entry) {
+            if (duration_parse (&duration, entry, &e) < 0) {
+                errprintf (error, "queues.%s.%s", name, e.text);
+                goto error;
+            }
+            if (queues_insert (zh, name, duration) < 0) {
+                errprintf (error, "out of memory parsing [queues]");
+                goto error;
+            }
+        }
+    }
+    *zhp = zh;
+    return 0;
+error:
+    ERRNO_SAFE_WRAP (zhashx_destroy, &zh);
+    return -1;
+}
+
+static int check_limit (struct limit_duration *ctx,
+                        double duration,
+                        const char *queue,
+                        flux_error_t *error)
+{
+    if (duration != DURATION_UNLIMITED) {
+        double limit = ctx->general_limit;
+        double qlimit = queues_lookup (ctx->queues, queue);;
+
+        if (qlimit != DURATION_INVALID)
+            limit = qlimit;
+        if (limit != DURATION_INVALID
+            && limit != DURATION_UNLIMITED
+            && duration > limit) {
+            char fsd[64];
+            fsd_format_duration_ex (fsd, sizeof (fsd), limit, 1);
+            return errprintf (error,
+                              "requested duration exceeds policy limit of %s",
+                              fsd);
+        }
+    }
+    return 0;
+}
+
+static int validate_cb (flux_plugin_t *p,
+                        const char *topic,
+                        flux_plugin_arg_t *args,
+                        void *arg)
+{
+    struct limit_duration *ctx = flux_plugin_aux_get (p, auxkey);
+    flux_job_state_t state;
+    double duration = DURATION_UNLIMITED;
+    const char *queue = NULL;
+    flux_error_t error;
+
+    if (flux_plugin_arg_unpack (args,
```

Comment:
any reason not to have the same comments / checks as in the other limits plugin?

```


    /* If no limits are configured, return immediately.  This is the common
     * case for a non-system instance and since this plugin is always loaded,
     * don't waste time.
     */
    if (!limits_isset (&ctx->general_limits)
        && zhashx_size (ctx->queues) == 0)
        return 0;
```

and

```
    /* Parse (optional) jobspec attributes.system.queue.
     * Leave queue NULL if unspecified.
     * Throw an error if it's the wrong type or related.
     */
```